### PR TITLE
Feature/29696

### PR DESCRIPTION
**Pull request by kjugi**
+ Es-lint was added to dependencies + updated in `package.json`
+ Sass-lint was added to dependencies + updated in `package.json`
+ Style-lint was added to dependencies + updated in `package.json`
+ Stylelin & Esling config added + updated in `package.json`
+ Few necessery dependencies installed and existing in `package.json`

Added gulp lints task and tested

Added all config and ignore files + Travis CI file

Repaired all errors in components.

https://dev.snowdog.pro/issues/29696

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,10 +4,22 @@
   "dependencies": {
     "@frctl/fractal": "^1.1.1",
     "autoprefixer": "^6.7.2",
+    "eslint-config-idiomatic": "^3.1.0",
```
Comment (Igloczek) [inline package.json]: Let's use `~` for every package.

File: package.json [hunk 2]
```diff
@@ -4,10 +4,22 @@
...
+    "gulp-sass-lint": "~1.3.2",
+    "gulp-sourcemaps": "^2.4.1",
+    "gulp-util": "~3.0.8",
+    "postcss-reporter": "~3.0.0",
+    "run-sequence": "~1.2.2",
+    "stylelint": "~7.10.1",
+    "stylelint-config-standard": "~16.0.0"
+  },
+  "scripts": {
+    "test": "./node_modules/eslint/bin/eslint.js *.js helper/*.js task/*.js"
```
Comment (Igloczek) [inline package.json]: Remove this one and add `gulp js-lint --ci` to the Travis config.

File: gulpfile.js [hunk 3]
```diff
@@ -25,158 +38,194 @@ fractal.docs.set('path', `${__dirname}/docs`); // location of the documentation
...
-        gulp.src('docs/styles.scss')
-            .pipe(plumber())
-            .pipe(sourcemaps.init())
-            .pipe(sass({outputStyle: 'compressed'}))
-            .pipe(postcss(processors))
-            .pipe(sourcemaps.write(paths.maps))
-            .pipe(gulp.dest(paths.css))
-    })
+  gulp.watch('**/*.scss', () => {
+    gulp.src(paths.scss)
```
Comment (Igloczek) [inline gulpfile.js]: Pointless code, check what you are running in line 191.

File: package.json (hunk 2 above)
Comment (Igloczek) [inline package.json]: Remember to not copy paste code blindly, b/c in this case you don't even have `helper` and `task` directory in project 😉

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,29 @@
+extends:
+  - eslint:recommended
+  - idiomatic
+env:
+  browser: true
+  node: true
```
Comment (Igloczek) [inline .eslintrc]: Remove this one.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+node_js: 6
+cache:
+  yarn: true
+  directories:
+    - node_modules
+script:
+  - yarn test
+  - gulp sass-lint
+  - gulp js-lint
+  - gulp css-lint
```
Comment (Igloczek) [inline .travis.yml]: Add `--ci` flags.

File: .eslintrc (hunk 4 above)
Comment (kjugi) [inline .eslintrc]: removed

File: .travis.yml (hunk 5 above)
Comment (kjugi) [inline .travis.yml]: `--ci` flags added

File: gulpfile.js (hunk 3 above)
Comment (kjugi) [inline gulpfile.js]: This is the same, so i deleted this one

File: package.json (hunk 1 above)
Comment (kjugi) [inline package.json]: used

File: package.json (hunk 2 above)
Comment (kjugi) [inline package.json]: You mean to delete this `test` script and delete from `travis.yml` file line with `yarn test` or left only `yarn`?

Comment (Igloczek) [inline package.json]: `yarn` is executed by default. If you remove `test` from `scripts`, you have to remove it everywhere 😉

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+language: node_js
+node_js: 6
+cache:
+  yarn: true
+  directories:
+    - node_modules
+script:
+  - yarn test
```
Comment (Igloczek) [inline .travis.yml]: `yarn` is executed by default. If you remove `test` from `scripts`, you have to remove it everywhere 😉

**Comment by Igloczek** [comment]
Remember to resolve conflicts.

File: .travis.yml (hunk 6 above)
Comment (kjugi) [inline .travis.yml]: `test` removed and flag `--ci` added

File: .eslintrc [hunk 7]
```diff
@@ -3,7 +3,6 @@ extends:
   - idiomatic
 env:
   browser: true
-  node: true
```
Comment (kjugi) [inline .eslintrc]: When this is deleted it generates errors with `'module' is not defined` in each `config.js` file

Comment (kjugi) [inline .eslintrc]: When this is deleted it generates errors with `'module' is not defined` in each `config.js` file

**Pull request merged**